From: Philipp Matthias Hahn <pmhahn@debian.org>
Subject: Bug#599507: SVM: Fix wrong intercept masks on 32 bit

The bug is not fixed, since
features/all/KVM-x86-Add-KVM_GET-SET_VCPU_EVENTS.patch from
debian/pacthes/series/12 introduces the new function svm_set_nmi_mask(),
which also has the wrong "1UL" instead of the correct "1ULL".

--- a/arch/x86/kvm/svm.c        2010-11-24 14:29:05.000000000 +0100
+++ b/arch/x86/kvm/svm.c        2010-11-24 13:24:35.000000000 +0100
@@ -2566,10 +2479,10 @@ static void svm_set_nmi_mask(struct kvm_
 
 	if (masked) {
 		svm->vcpu.arch.hflags |= HF_NMI_MASK;
-		svm->vmcb->control.intercept |= (1UL << INTERCEPT_IRET);
+		svm->vmcb->control.intercept |= (1ULL << INTERCEPT_IRET);
 	} else {
 		svm->vcpu.arch.hflags &= ~HF_NMI_MASK;
-		svm->vmcb->control.intercept &= ~(1UL << INTERCEPT_IRET);
+		svm->vmcb->control.intercept &= ~(1ULL << INTERCEPT_IRET);
 	}
 }
 
